File: python/linear_regression.py
# Rafael Henriques
# 15-Jul-2019

import numpy as np # linear algebra
import pandas as pd # data processing
import matplotlib.pyplot as plt

# Read Dataset
df = pd.read_csv("../dataset/iris.data")

print(df.shape)
print(df.info())
print(df.head(4));

# Remove Unacessary Data
df = df[['petalLength','petalWidth']].copy()
print(df.head(2))

# Prepare the training and testing data
# 4/5 (80%) train
# 1/5 (20%) test
from sklearn.model_selection import train_test_split
x = df.iloc[:,0:1] # feature # ending index is exclusive
y = df.iloc[:,1:2] # target
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)

# Linear Regression
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(x_train, y_train)

# Predict / Test
y_predict = model.predict(x_test)

# Plot
plt.plot(x_test,y_predict)
plt.scatter(x_test,y_predict)
plt.scatter(x_test,y_test, color="orange")
plt.show()

File: R/linear_regression.r
# Rafael Henriques
# 15-Jul-2019

# Read Dataset
df = read.table("../dataset/iris.data", header=TRUE, sep=",")

summary(df)

head(df)

# Correlation Matrix
library(GGally)
ggpairs(df)

# Remove unnecessary columns (features)
df$sepalLength = NULL
df$sepalWidth = NULL



# Prepare the training and testing data
# 4/5 (80%) train
# 1/5 (20%) test
testIdx = which(1:length(df[,1]) %% 5 == 0)

dfTrain = df[-testIdx,]
dfTest = df[testIdx,]

summary(dfTrain)
summary(dfTest)

# Linear Regression
model = lm(formula=petalWidth ~ petalLength, data=dfTrain)

# Use the model to predict the output of test data
prediction = predict(model,newdata=dfTest)
summary(prediction)

cor(prediction, dfTest$petalWidth)
summary(model)

# Petal width dependent on sepal width
plot(x=dfTrain$petalLength, y=dfTrain$petalWidth, main="petalWidth ~ petalLength")
# Regression linear function
abline(model)


